src/mscorlib/src/System/ThrowHelper.cs

```diff
@@ -119,323 +189,34 @@ internal static class ThrowHelper {
-                    break;
-
-                case ExceptionArgument.count:
-                    argumentName = "count";
-                    break;
-
-                case ExceptionArgument.dictionary:
-                    argumentName = "dictionary";
-                    break;
-
-                case ExceptionArgument.dictionaryCreationThreshold:
-                    argumentName = "dictionaryCreationThreshold";
-                    break;
-
-                case ExceptionArgument.index:
-                    argumentName = "index";
-                    break;
-
-                case ExceptionArgument.info:
-                    argumentName = "info";
-                    break;
-
-                case ExceptionArgument.key:
-                    argumentName = "key";
-                    break;
-
-                case ExceptionArgument.match:
-                    argumentName = "match";
-                    break;
-
-                case ExceptionArgument.obj:
-                    argumentName = "obj";
-                    break;
-
-                case ExceptionArgument.queue:
-                    argumentName = "queue";
-                    break;
-
-                case ExceptionArgument.stack:
-                    argumentName = "stack";
-                    break;
-
-                case ExceptionArgument.startIndex:
-                    argumentName = "startIndex";
-                    break;
-
-                case ExceptionArgument.value:
-                    argumentName = "value";
-                    break;
-
-                case ExceptionArgument.name:
-                    argumentName = "name";
-                    break;
-
-                case ExceptionArgument.mode:
-                    argumentName = "mode";
-                    break;
-
-                case ExceptionArgument.item:
-                    argumentName = "item";
-                    break;
-
-                case ExceptionArgument.options:
-                    argumentName = "options";
-                    break;
-
-                case ExceptionArgument.view:
-                    argumentName = "view";
-                    break;
-
-               case ExceptionArgument.sourceBytesToCopy:
-                    argumentName = "sourceBytesToCopy";
-                    break;
-
-                default:
-                    Contract.Assert(false, "The enum value is not defined, please checked ExceptionArgumentName Enum.");
-                    return string.Empty;
+        internal static string GetArgumentName(ExceptionArgument argument)
+        {
+            // box once
+            Enum arg = argument;
+            if (!Enum.IsDefined(typeof(ExceptionArgument), arg))
+            {
+                Contract.Assert(false, "The enum value is not defined, please check the ExceptionArgument Enum.");
+                return string.Empty;
             }
 
-            return argumentName;
+            return arg.ToString();
```

Comment:
Could the `Enum.IsDefined` check be removed from release builds? Why not assert as follows?

```c#
internal static string GetArgumentName(ExceptionArgument argument)
{
    Contract.Assert(Enum.IsDefined(typeof(ExceptionArgument), argument),
        "The enum value is not defined, please check the ExceptionArgument Enum.");

    return argument.ToString();
}
```